### PR TITLE
[COR-502] adding in a chunker for handling operations that should be …

### PR DESCRIPTION
**Pull request by apynes**
…batched

### CODE REVIEW

File: messagequeue/chunker.go [hunk 1]
```diff
@@ -40,10 +40,10 @@ type ChunkerOptions struct {
 // Valid returns an error is a options value is out of bounds.
 func (o *ChunkerOptions) Valid() error {
 	if o.Size == 0 {
-		return errors.New("ChunkSize must be greater thank 0")
+		return errors.New("Size must be greater thank 0")
```
Comment (webpro-jjaynes) [inline messagequeue/chunker.go:43]: I think you already fixed this in the other PR, but it should be "than 0"

File: messagequeue/chunker.go [hunk 2]
```diff
@@ -40,10 +40,10 @@ type ChunkerOptions struct {
 // Valid returns an error is a options value is out of bounds.
```
Comment (webpro-jjaynes) [inline messagequeue/chunker.go:40]: "..error if an options value..."

**Review by webpro-jjaynes — CHANGES_REQUESTED** [review]

File: messagequeue/chunker.go [hunk 3]
```diff
@@ -37,14 +40,16 @@ type ChunkerOptions struct {
...
+		return errors.New("ChunkerOptions.Size(%d) was out of bounds [%d, %d]",
+			o.Size, minimumChunkSize, maximumChunkSize,
+		)
 	}
-	if o.ElementExpiry < minimumExpiry {
-		return errors.New("ChunkElementExpiry must be greater than %s",
-			minimumExpiry.String())
+	if o.ElementExpiry < minimumExpiry || o.ElementExpiry > maximumExpiry {
+		return errors.New("ChunkerOptions.ElementExpiry(%s) was out of bounds [%s,%s]",
+			minimumExpiry.String(), minimumExpiry, maximumExpiry)
```
Comment (anthonyheidenreich) [inline messagequeue/chunker.go:52]: `minimumExpiry.String()` should be `o.ElementExpiry`

**Review by anthonyheidenreich — CHANGES_REQUESTED** [review]

**Review by morus12 — APPROVED** [review]

File: messagequeue/enqueuer.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+func (qr *enqueuer) Enqueue(message *Message) error {
+	if !qr.Running() {
+		return errors.New("cannot enqueue into a stopped MessageQueue")
+	}
+	// this will block if the buffer is full
+	qr.buffer <- message
+	return nil
+}
+
+func (qr *enqueuer) logFailed(nqr Enqueuer, result *EnqueueMessageResult) {
```
Comment (anthonyheidenreich) [inline messagequeue/enqueuer.go:115]: Why is this taking in the Enqueuer if it's not using it?

**Review by anthonyheidenreich — APPROVED** [review]

File: messagequeue/chunker.go [hunk 5]
```diff
@@ -74,32 +43,36 @@ type Handler[T any] func(chunk []T)
...
 	}
+	c.buffer = buffer
+	c.handler = handler
 	// validate our options
-	if err := c.options.Valid(); err != nil {
+	if err := c.options.Validate(); err != nil {
 		return err
 	}
+	c.buffer = buffer
+	c.handler = handler
```
Comment (nmarie22) [inline messagequeue/chunker.go:75]: Is there a reason you are setting buffer and handler again?

**Review by nmarie22 — APPROVED** [review]

File: messagequeue/enqueuer.go (hunk 4 above)
Comment (apynes) [reply messagequeue/enqueuer.go:115]: this is actually a satisfying the func definition for error handler. A typical handler would re-enqueue, but when one is not provided this is used to just log it and throw it away.

File: messagequeue/chunker.go (hunk 5 above)
Comment (apynes) [reply messagequeue/chunker.go:75]: refactor artifact. good catch :+1:

**Review by webpro-jjaynes — APPROVED** [review]

**Pull request merged**